README.md

```diff
@@ -1598,10 +3338,203 @@ end
+```
+render() {
+  return (
+    <div className="container">
+      <NewSkill handleSubmit={this.handleSubmit} />
+      <SelectFilter handleFilter={this.filterSkillsByLevel} />
+      <AllSkills skills={this.state.skills}
+                 handleDelete={this.handleDelete}
+                 handleUpdate={this.handleUpdate} />
+      </div>
+  )
+}
+```
+<br>
+I am sure you noticed that inside of our `handleFilter` property we have a function called `filterSkillsByLevel`, so let's go ahead and create that within our `_body.js.jsx`.
+
+**app/assets/components/javascripts/_body.js.jsx**
+```
+filterSkillsByLevel(level) {
+  console.log("about to filter");
+},
+```
+We will add the functionality of filtering in the above `filterSkillsByLevel(level)` function. But first, let's wire it all together and make sure that when the user changes the select dropdown, this `filterSkillsByLevel` function is invoked.
+
+
+Let's add an `onChange` event to our select input that will call the `updateFilter` function. In order to accomplish that we must bind the object 'this' to our function call. In this case, the 'this' object refers to the `SelectFilter` component. This is important because if we did not bind 'this', when we get into the updateFilter function call we would not be able to call `this.state` or `this.props` since in that case, this would refer to the function `updateFilter`, not the component `SelectFilter`. That is a lot of this's and thats and for what its worth, this is probably the toughest part of React!
+
+**app/assets/components/javascripts/_select_filter.js.jsx**
+```
+render (){
+  return(
+    <div>
+      <label>
+        Filter By
+      </label>
+      <select className="form-control" onChange={this.updateFilter.bind(this)}>  
+    
+    ...
+```
+Now we need to create an updateFilter function within the FilterSelect component which will be invoked whenever there is a change on our element.
+
+**app/assets/components/javascripts/_select_filter.js.jsx**
+```
+updateFilter(event) {
+  this.props.handleFilter(event.target.value)
+}
+```
+Remember, since we bound the object 'this' to the function call in our `updateFilter`, 'this' refers to the entire component. This gives us access to all of the properties of the `SelectFilter` component. We need the `handleFilter` property, which we pass into the component when we render it in the Body. The `handleFilter` property points to the `filterSkillsByLevel` function in the `Body` component. 
+```
+         Main
+       /      \
+  Header        Body
+            /      \                 \ 
+        NewSkill   SelectFilter    AllSkills
+                                       \
+                                      Skills * n
+```
+For additioanl context, in the above diagram when the user selects a value to filter by in the `SelectFilter` component, the `handleFilter` property is invoked which sends a message up to the `body` component, which in turn re-renders the list of visible skills.
```

Comment:
typo: `additional`

Maybe also specify that when we send a message up to the body component, we perform a state change which is passed down to SelectFilter, and the state change is what would cause the re-render.